sdk/v2/builder.go

```diff
@@ -0,0 +1,116 @@
+package v2
+
+import (
+	"io"
+	"os"
+	"reflect"
+	"sync"
+
+	"github.com/newrelic/infra-integrations-sdk/args"
+	"github.com/newrelic/infra-integrations-sdk/cache"
+	"github.com/pkg/errors"
+)
+
+const protocolVersion = "2"
+
+// IntegrationBuilder provides a fluent interface for creating a configured Integration.
+type IntegrationBuilder interface {
+	// Build returns the Integration resulting from the applied configuration on this builder.
+	// The integration data is empty, ready to add new entities' data.
+	Build() (*Integration, error)
+	// ParsedArguments sets the destination struct (pointer) where the command-line flags will be parsed to.
+	ParsedArguments(interface{}) IntegrationBuilder
+	// Synchronized sets the built Integration ready to be managed concurrently from multiple threads.
+	// By default, the build integration is not synchronized.
+	Synchronized() IntegrationBuilder
+	// Writer sets the output stream where the integration resulting payload will be written to.
+	// By default, the standard output (os.Stdout).
+	Writer(io.Writer) IntegrationBuilder
+}
+
+type integrationBuilderImpl struct {
+	integration *Integration
+	arguments   interface{}
+}
+
+type disabledLocker struct{}
+
+func (disabledLocker) Lock()   {}
+func (disabledLocker) Unlock() {}
+
+// NewIntegration creates a new IntegrationBuilder for the given integration name and version.
+func NewIntegration(name string, version string) IntegrationBuilder {
+	return &integrationBuilderImpl{
+		integration: &Integration{
+			Name:               name,
+			ProtocolVersion:    protocolVersion,
+			IntegrationVersion: version,
+			Data:               []*EntityData{},
```

Comment:
I'd create a NewData constructor to be called from here responsible of EntityData fields initialization (ie make Inventory map)